### PR TITLE
fix: span names 

### LINKED ISSUE #542: Fix span names
Fix span names that cannot be updated when jwt auth happens and jwk result is cached

### PR DESCRIPTION
**Pull request by alekseyvdovenko**
Closes https://github.com/epam/ai-dial-core/issues/542

### CODE REVIEW

**Comment by alekseyvdovenko** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-543.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by alekseyvdovenko** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-543.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by akurnosau — DISMISSED** [review]

File: server/src/main/java/com/epam/aidial/core/server/security/IdentityProvider.java [hunk 1]
```diff
@@ -182,7 +182,8 @@ public static DecodedJWT decodeJwtToken(String encodedToken) {
     }
 
     private Future<JwkResult> getJwk(String kid) {
-        return cache.computeIfAbsent(kid, key -> vertx.executeBlocking(() -> {
+        Promise<JwkResult> promise = Promise.promise();
+        cache.computeIfAbsent(kid, key -> vertx.executeBlocking(() -> {
```
Comment (astsiapanay) [inline server/src/main/java/com/epam/aidial/core/server/security/IdentityProvider.java:191]: Please add a comment why we don’t reuse the cached result.

File: server/src/main/java/com/epam/aidial/core/server/Proxy.java [hunk 2]
```diff
@@ -280,6 +281,11 @@ private void respond(HttpServerRequest request, HttpStatus status) {
     }
 
     private void respond(HttpServerRequest request, HttpStatus status, String body) {
+        if (status == HttpStatus.OK) {
+            SpanUtil.updateName(request.method().name(), request.path());
+        } else if (status.is4xx() || status.is5xx()) {
```
Comment (astsiapanay) [inline server/src/main/java/com/epam/aidial/core/server/Proxy.java]: Why not use if/else?

what if http 3xx comes?

File: server/src/main/java/com/epam/aidial/core/server/Proxy.java [hunk 3]
```diff
@@ -280,6 +281,11 @@ private void respond(HttpServerRequest request, HttpStatus status) {
     }
 
     private void respond(HttpServerRequest request, HttpStatus status, String body) {
+        if (status == HttpStatus.OK) {
+            SpanUtil.updateName(request.method().name(), request.path());
```
Comment (astsiapanay) [inline server/src/main/java/com/epam/aidial/core/server/Proxy.java]: Why do we update span name again? Moreover we set a raw path.

So we do have different span names based on http codes. Why?

File: server/src/main/java/com/epam/aidial/core/server/util/SpanUtil.java [hunk 4]
```diff
@@ -12,7 +12,15 @@ public class SpanUtil {
     private static final List<String> GROUPS = List.of("id", "bucket", "path");
 
     public static void updateName(Pattern pathPattern, String path, String httpMethod) {
-        String spanName = RegexUtil.replaceNamedGroups(pathPattern, path, GROUPS);
-        Span.current().updateName(httpMethod + " " + spanName);
+        String httpPath = RegexUtil.replaceNamedGroups(pathPattern, path, GROUPS);
```
Comment (astsiapanay) [inline server/src/main/java/com/epam/aidial/core/server/util/SpanUtil.java]: We can do code re-factoring and move the methods to SelectorController class.

So the class is no longer needed in this case.

Comment (alekseyvdovenko) [reply server/src/main/java/com/epam/aidial/core/server/util/SpanUtil.java]: Will be fixed in scope of a separate issue

File: server/src/main/java/com/epam/aidial/core/server/Proxy.java (hunk 3 above)
Comment (alekseyvdovenko) [reply server/src/main/java/com/epam/aidial/core/server/Proxy.java]: Will be fixed in scope of a separate issue

File: server/src/main/java/com/epam/aidial/core/server/Proxy.java (hunk 2 above)
Comment (alekseyvdovenko) [reply server/src/main/java/com/epam/aidial/core/server/Proxy.java]: Will be fixed in scope of a separate issue

File: server/src/main/java/com/epam/aidial/core/server/security/IdentityProvider.java (hunk 1 above)
Comment (alekseyvdovenko) [reply server/src/main/java/com/epam/aidial/core/server/security/IdentityProvider.java:191]: added

**Review by astsiapanay — APPROVED** [review]

**Pull request merged**